screen-weather-get.py

```diff
@@ -1,200 +1,69 @@
+configure_logging()
 
-    url = ("https://data.climacell.co/v4/timelines"
-        + "?location={}&units={}&fields=temperatureMin&fields=temperatureMax&fields=weatherCode&timesteps=1d&apikey={}"
-        .format(location_latlong, units, climacell_apikey))
-    try:
-        response_data = get_response_data(url, os.getcwd() + "/" + filename, ttl)
-        weather = response_data['timelines'][0]['intervals'][0]['values']
-        logging.debug("get_weather() - {}".format(weather))
-    except Exception as error:
-        logging.error(error)
-        weather = False
+def get_cached_weather(filename, ttl):
+    if is_stale(filename, ttl):
+        logging.info("Weather cache is stale.")
+        return None
 
-    return weather
+    logging.info("Found in cache")
+    with open(filename, 'r') as file:
+        return json.loads(file.read())
+    
 
-def get_response_data(url, filepath, ttl):
+def cache_weather_data(filename, weather_data):
+    if weather_data:
+        with open(filename, 'w') as text_file:
+            json.dump(weather_data, text_file)
 
-    response_json = False
-
-    if (is_stale(filepath, ttl)):
-        try:
-            response_data = requests.get(url).text
-            response_json = json.loads(response_data)['data']
-            with open(filepath, 'w') as text_file:
-                text_file.write(response_data)
-        except Exception as error:
-            logging.error(error)
-            raise
-    else:
-        with open(filepath, 'r') as file:
-            return json.loads(file.read())['data']
-    return response_json
 
 def main():
 
     # gather relevant environment configs
-
-    climacell_apikey=os.getenv("CLIMACELL_APIKEY","")
-
-    if climacell_apikey=="":
-        logging.error("CLIMACELL_APIKEY is missing")
+    climacell_apikey=os.getenv("CLIMACELL_APIKEY")
+    openweathermap_apikey=os.getenv("OPENWEATHERMAP_APIKEY")
+    
+    if not climacell_apikey and not openweathermap_apikey:
+        logging.error("API Key for weather is missing (Climacell, OpenWeatherMap)")
         sys.exit(1)
 
     weather_format=os.getenv("WEATHER_FORMAT", "CELSIUS")
-
     if (weather_format == "CELSIUS"):
         units = "metric"
     else:
         units = "imperial"
 
-    template_svg_filename = 'screen-template.svg'
-    output_svg_filename = 'screen-output-weather.svg'
-
-    # json response files
-    weather_timelines_filename = 'climacell-timelines-response.json'
-
-    location_lat = os.getenv("WEATHER_LATITUDE","51.3656") 
-    location_long = os.getenv("WEATHER_LONGITUDE","-0.1963") 
-
-    location_latlong = (
-        "{0:.2f},{1:.2f}"
-        .format(float(location_lat), float(location_long)))
+    location_lat = os.getenv("WEATHER_LATITUDE","51.3656")
+    location_long = os.getenv("WEATHER_LONGITUDE","-0.1963")
 
     # TTL for refetching of JSON
     ttl = float(os.getenv("WEATHER_TTL", 1 * 60 * 60))
+    cache_weather_file = "weather-cache.json"
+
+    if openweathermap_apikey:
```

Comment:
Not going to do fancy OOP, just a bunch of if statements.  Based on whatever environment variables, the corresponding module will be called.